src/components/Spinner/Spinner.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { Fragment, useEffect, useState } from 'react';
+import styles from './Spinner.scss';
+
+const timeoutIntervalMs = 10000;
+
+const Spinner = ( props: {
+	/**
+	 * Whether to display a timeout message to the user after a number of seconds have elapsed.
+	 */
+	timeout?: boolean,
+} ) => {
+	const { timeout } = props;
```

Comment:
this would let the timeout be configurable

```suggestion
	const { timeout = 0 } = props;
```